### ISSUE TITLE
Create a configured room (10.1.3)

### ISSUE THREAD

**Issue by Cris2760** [coyim/coyim#639]
A room can have a lot of configuration options to set. Exactly what these are depends on the server and how it's configured - you will basically get a form to fill in. However, most other clients will show the form directly. This isn't necessarily great for usability, so for this story I think we should break up the different configuration options into different areas, or tabs, or something else that helps the user. We should show the most important or most common options first, and then the more advanced or weird ones. We also need to provide some kind of explanations and help for these.

This story also needs to include validations of all the options and helpful feedback if you don't give the right values.

Basically, this story extends the simple creation of a room, allowing you to add configuration options. We should probably _not_ add a new menu option to access this, but instead allow you to add configuration options from the basic room creation dialog.

**Note:** Please see the reference on https://xmpp.org/extensions/xep-0045.html

**Comment by olabiniV2** [comment]
This is part of #271.

**Comment by peterpalau** [comment]
Okay as we agreed this issue is ready for user testing.

**Comment by peterpalau** [comment]
We think the issue is ready to be reviewed.

**Comment by peterpalau** [comment]
I noticed that a comment was missing from the user tests.

**Comment by peterpalau** [comment]
This should go first to user testing in order to check if everything is good.

**Comment by peterpalau** [comment]
The user testing is done. The issue is ready to be reviewed.

**Comment by olabiniV2** [comment]
I have a big question about this issue. Does it support other, unknown, settings? That was definitely one of the design goals.

**Comment by peterpalau** [comment]
> I have a big question about this issue. Does it support other, unknown, settings? That was definitely one of the design goals.

And definitely, this is a really good question. This issue doesn't support other/unknown/dynamic settings.

**Comment by olabiniV2** [comment]
Well, the original requirements for this story did include the facilitity to manage unknown configuration values. As I described it in numerous meetings, the idea was that we get the form from the server, we provide a nice experience for all the configuration options we know about, and then display the rest in some default way.

So, that means that this story is not even close to done. I will move it back to "Doing" for now, for that work to start. In the meantime, I'll continue reviewing all the old commits.

**Comment by olabiniV2** [comment]
I have now reviewed all the commits up to this point. There are several things to fix.

**Comment by olabiniV2** [comment]
OK

**Comment by Cris2760** [comment]
We decided reopen this issue because it remains to verify the server error messages and manage them so that they are user-friendly.

**Comment by azlancpool** [comment]
Currently we have a problem with the notification bar in room configuration assistant, but it will be handled within the notification bar component so, it's ready for user-test

**Comment by peterpalau** [comment]
After the user tests, we agreed that the navigation buttons of the configuration assistant window should be disabled if any error is present on the current configuration page.

**Comment by olabiniV2** [comment]
Looks good.

**Comment by peterpalau** [comment]
The issue is ready to be closed.
